packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -6,372 +6,179 @@
-    ) private view {
-        // Check anchor tx is valid
-        LibTxDecoder.Tx memory _tx = LibTxDecoder.decodeTx(
-            config.chainId,
-            anchorTx
-        );
-        if (_tx.txType != 0) revert L1_ANCHOR_TYPE();
-        if (_tx.destination != resolver.resolve(config.chainId, "taiko", false))
-            revert L1_ANCHOR_DEST();
-        if (_tx.gasLimit != config.anchorTxGasLimit)
-            revert L1_ANCHOR_GAS_LIMIT();
-
-        if (FLAG_VALIDATE_ANCHOR_TX_SIGNATURE) {
-            // Check anchor tx's signature is valid and deterministic
-            if (
-                _tx.r != LibAnchorSignature.GX &&
-                _tx.r != LibAnchorSignature.GX2
-            ) revert L1_ANCHOR_SIG_R();
+                false
+            );
+
+            bytes32 instance;
+            {
+                // otherwise: stack too deep
+                address l1SignalService = resolver.resolve(
+                    "signal_service",
+                    false
+                );
+                address l2SignalService = resolver.resolve(
+                    config.chainId,
+                    "signal_service",
+                    false
+                );
 
-            if (_tx.r == LibAnchorSignature.GX2) {
-                (, , uint256 s) = LibAnchorSignature.signTransaction(
-                    LibTxUtils.hashUnsignedTx(config.chainId, _tx),
-                    1
+                bytes memory buffer = bytes.concat(
+                    // for checking anchor tx
+                    bytes32(uint256(uint160(l1SignalService))),
+                    // for checking signalRoot
+                    bytes32(uint256(uint160(l2SignalService))),
+                    evidence.parentHash,
+                    evidence.blockHash,
+                    evidence.signalRoot
+                );
+                buffer = bytes.concat(
+                    buffer,
+                    bytes32(uint256(uint160(evidence.prover))),
+                    bytes32(uint256(evidence.meta.id)),
+                    bytes32(evidence.meta.l1Height),
+                    evidence.meta.l1Hash,
+                    evidence.meta.txListHash
                 );
-                if (s != 0) revert L1_ANCHOR_SIG_S();
+
+                instance = keccak256(buffer);
             }
-        }
 
-        // Check anchor tx's calldata is valid
-        if (
-            !LibBytesUtils.equal(
-                _tx.data,
+            (bool verified, bytes memory ret) = verifier.staticcall(
                 bytes.concat(
-                    ANCHOR_TX_SELECTOR,
-                    bytes32(evidence.meta.l1Height),
-                    evidence.meta.l1Hash
+                    bytes16(0),
+                    bytes16(instance), // left 16 bytes of the given instance
+                    bytes16(0),
+                    bytes16(uint128(uint256(instance))), // right 16 bytes of the given instance
+                    evidence.zkproof.data
```

Comment:
Had a comment here https://github.com/taikoxyz/taiko-mono/pull/13280#discussion_r1131326540, just making sure it isn't lost.
> I think data like timestamp and mixhash are missing here so the circuit can verify the blockhash is calculated using these expected values. Or could also be I'm overlooking where these are inserted in the circuit.

Maybe best not to split up `instance` here already, that seems verifier implementation dependent.

Reply:
@davidtaikocha @smtmfft what do you think?

Reply:
The new pi-circuit design uses a hash value as public input, but we use a finite field with a maximum of 253 bits, the length of the hash is 256 bits, so we split it into two parts.

I'm not sure if my answer is correct for here.